src/Autofac.Integration.AspNetCore.Multitenant/MultitenantRequestServicesMiddleware.cs

```diff
@@ -11,6 +11,13 @@
 
 namespace Autofac.Integration.AspNetCore.Multitenant
 {
+    /// <summary>
+    /// The factory used to retrieve the <see cref="IServiceScopeFactory"/>.
+    /// </summary>
+    /// <param name="lifetimeScope">The <see cref="ILifetimeScope"/> which can either be the root scope or dedicated tenant-scope.</param>
+    /// <returns>The <see cref="IServiceScopeFactory"/> for the specific <see cref="ILifetimeScope"/> which can either be the root scope or dedicated tenant-scope.</returns>
+    internal delegate IServiceScopeFactory ServiceScopeFactory(ILifetimeScope lifetimeScope);
```

Comment:
I wonder if there's a better name for this since there are already so many of these overloaded "service scope factory provider accessor" sorts of things. I don't have suggestions, and I'm not going to block the PR over it, just thinking out loud. I mean, it's internal, so we can always change it later without breaking the API.

Reply:
Yeah it is kind of confusing. I can only think of really long names. 

What if we were to remove `Factory` and use the term `Accessor` here instead?

Reply:
I mean, technically it's _getting an `IServiceScopeFactory`_ so it's a `ServiceScopeFactoryAccessor`. Which isn't better, but is probably better than `ServiceScopeFactoryFactory`. We could go totally astray and call it `GetServiceScopeFactory` to explain what it _does_, which is... uh... it... gets service scope factories. I'm not in love with any of these. 😦

Reply:
I renamed it to `ServiceScopeFactoryAccessor` for now. We can change it at anytime since it is internal and can only be accessed by us (exception of reflection stuff).